### PR TITLE
Selectable search marker icon

### PR DESCRIPTION
**Pull request by Becold**
## Description
The user can change the icon marker if he dont like it.
Changed the default icon marker to the [Google's one](http://stackoverflow.com/questions/25368851/google-maps-default-icons).

The marker styles are located in static/data/searchmarkerstyle.json

## Motivation and Context
A lot of people hate the new Ash marker from commit 8080e8edc026e19248c1a759522591e6cf5d1173 
And https://github.com/PokemonGoMap/PokemonGo-Map/issues/697 asked my "Why is there a Mario on the map ? lol" 

## How Has This Been Tested ?
Deleting my cache and my localstorage between each test.
See screenshots

## Screenshots :
On the front-end : 
image

image

image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by invisiblek** [comment]
I think it makes more sense for this to be handled on the front end. Users could select what they want and store their setting in localStorage.

**Comment by Studio-XP** [comment]
Yes, I agree that it should be handled by the front end. I've started putting something together last night in map.js. I'll put it up later on when I get off work

**Comment by Becold** [comment]
Should I had on the frontend (and keep it on the backend)
Or I close this PR and only do something on the frontend ?

**Comment by jnovack** [comment]
Front-end only please.

**Comment by Becold** [comment]
I'm done with the front-end.
The back-end is good to keep. It's by default on ash (as it is now).

**Comment by Albert62** [comment]
Here is my icon.
http://i.imgur.com/MarpyYl.png

Preview:
http://i.imgur.com/YD8SllU.jpg

Enjoy.

**Comment by Moussa** [comment]
>Add an argument to change the original icon marker.

The original icon marker wasn't invisible though?

**Comment by Becold** [comment]
@Moussa  I'll add the [default icon](http://stackoverflow.com/questions/25368851/google-maps-default-icons) (from Google) to the list.

@invisiblek Which icon do I set as default? Ash or Google's icon marker ?

**Comment by invisiblek** [comment]
probably google's as the default

**Comment by Becold** [comment]
I edited the original PR.
Removed all the python/backend stuff (the argument -om is gone) and Keeped all the front-end/UI stuff (with the list).

File: templates/map.html [hunk 1]
```diff
@@ -210,6 +210,14 @@
                 <option value="20">X-Large</option>
               </select>
             </div>
+            <div class="form-control switch-container">
+              <h3>Icon Marker</h3>
+              <select name="iconmarker-style" id="iconmarker-style">
+                <option value="default">Default</option>
+                <option value="ash">Ash</option>
+                <option value="none">None</option>
+			  </select>
```
Comment (invisiblek) [inline templates/map.html]: should be indented the same as line 215

File: static/js/map.js [hunk 2]
```diff
@@ -761,6 +762,10 @@ var StoreOptions = {
   'iconSizeModifier': {
     default: 0,
     type: StoreTypes.Number
+  },
+  'originalMarker': {
```
Comment (invisiblek) [inline static/js/map.js]: I'm not sure this is really an accurate name for this variable. Maybe something like searchMarkerStyle or something?

File: templates/map.html [hunk 3]
```diff
@@ -210,6 +210,14 @@
                 <option value="20">X-Large</option>
               </select>
             </div>
+            <div class="form-control switch-container">
+              <h3>Icon Marker</h3>
```
Comment (invisiblek) [inline templates/map.html]: Same as in map.js, I'm not sure this is entirely descriptive of what it changes.

**Comment by Studio-XP** [comment]
Would it be better to load the values from a javascript object and load them into the select container?  For example:

```
var markerStyles = {
  google: {
    name: 'Default Google Marker',
    icon: '',
  },
  red: {
    name: 'Red',
    icon: 'static/markers/red.gif',
  },
  red_animated: {
    name: 'Red (Animated)',
    icon: 'static/markers/red_animated.gif',
  },
  none: {
    name: 'Hidden',
    icon: 'static/markers/blank.gif',
  }
}
```

**Comment by Becold** [comment]
@Studio-XP Done. But gif aren't working as icon marker andI dont know why... (Rip animated Red)

**Comment by Studio-XP** [comment]
Someone requested that I make a blue and yellow version too (losers, Team Valor ftw). I went with a static gif over the png because the png version isn't a true 8-bit version (zoom in on it). I also modified the color of Red

red
red_animated
yellow
yellow_animated
blue
blue_animated

**Comment by Becold** [comment]
They look cool, i'll add them for the next commit.
But do you know why the gif arent working ? They are not animated :cry:

**Comment by Studio-XP** [comment]
you have to add `optimized: false` in createSearchMarker()

**Comment by Becold** [comment]
@Studio-XP Thank you, you're the best

**Comment by Studio-XP** [comment]
Always glad to help.  Btw, noticed a small error

```
'searchMarkerStyle': {
    default: 'default',
    type: StoreTypes.String
```
should be this
```
'searchMarkerStyle': {
    default: 'google',
    type: StoreTypes.String
```

**Comment by chuyskywalker** [comment]
This looks fantastic.

**Comment by hklcf** [comment]
great

**Comment by Studio-XP** [comment]
I can redo the pokeball as an 8 bit too when I get home if you want. Looks blurry

**Comment by Becold** [comment]
Up to you @Studio-XP

**Comment by Studio-XP** [comment]
As promised
pokeball

**Comment by Becold** [comment]
Ready to be merged, I guess

**Comment by rrbailey89** [comment]
@becold on mobile the drop down menu is very tiny and can't be selected.

**Comment by Studio-XP** [comment]
Not sure why it's like that. Should be like the rest of the menus. Uses the same code. It's working as expected on chrome for Android.

**Comment by Becold** [comment]
@rrbailey89 Did you run `grunt build` and clear the cache ?

**Comment by Studio-XP** [comment]
Yeah, it almost sounds like grunt didn't build the search markers json file in dist/data. Are there any options?

If you're adding it without downloading it from github, you can technically just copy that file to that location

**Comment by rrbailey89** [comment]
@Becold @Studio-XP it works perfectly on my desktop.  Sorryou I'm not running the server atm to show the screen shot however on mobile for me the dropdown menu is present but it's width is shrunk down and when tapped displays no list

**Comment by chuyskywalker** [comment]
Works on iPhone for me.

**Comment by Becold** [comment]
@rrbailey89 What is your setup, did you have nginx/apache reverse proxy ?
If so, you might have to create a new route to static/dist/data/searchmarkerstyle.min.json (which is auto-generated by `grunt build`)

**Comment by rrbailey89** [comment]
@Becold my apologies for taking so long to get back to you. I simply execute runserver.py im not behind any proxy. This first Screenshot is what I see when using my desktop. 
ghriolg

this is what I see when using my Nexus 6.
screenshot_20160818-072605-2

I hope this helps.

**Comment by Zoruk** [comment]
You just have to clear browser cache and refresh

**Comment by rrbailey89** [comment]
@Zoruk @Becold Okay I guess it was the Cache :( .. never had to manually clear my devices browser cache before for this app... but after I did the dropdown menu populated properly.. [this site is relevant for this](http://shouldiblamecaching.com/)

**Pull request merged**